cdap-spark-core/src/main/scala/co/cask/cdap/app/runtime/spark/DatasetCompute.scala

```diff
@@ -35,5 +35,6 @@ private[spark] trait DatasetCompute {
     * @tparam T type of the result
     * @return result of the computation by the function
     */
-  def apply[T: ClassTag](datasetName: String, arguments: Map[String, String], computeFunc: Dataset => T): T
+  def apply[T: ClassTag](datasetNamespace: String, datasetName: String, arguments: Map[String, String],
```

Comment:
Since this is an inner class method, the change won't affect existing programs. When a datasetRDD is constructed now, it will fill the datasetNamespace with the namespace where program is running if it is not provided: https://github.com/caskdata/cdap/pull/6036/files#diff-a7787a2974e6d2a911ac39edfc262272R154

We also have the original unit test cases on this.